Sources/SwiftGen/Array+Parallel.swift

```diff
@@ -0,0 +1,29 @@
+//
+// SwiftGen
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+// Based on https://github.com/realm/SwiftLint/blob/0.39.2/Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift
+
+extension Array {
+  func parallelFlatMap<T>(transform: (Element) -> [T]) -> [T] {
+    parallelMap(transform: transform).flatMap { $0 }
+  }
+
+  func parallelCompactMap<T>(transform: (Element) -> T?) -> [T] {
+    parallelMap(transform: transform).compactMap { $0 }
+  }
+
+  func parallelMap<T>(transform: (Element) -> T) -> [T] {
+    var result = ContiguousArray<T?>(repeating: nil, count: count)
+    return result.withUnsafeMutableBufferPointer { buffer in
+      DispatchQueue.concurrentPerform(iterations: buffer.count) { idx in
+        buffer[idx] = transform(self[idx])
+      }
+      return buffer.compactMap { $0 }
```

Comment:
This is different to the SwiftLint code it's based on, which uses:
```swift
return buffer.map { $0! }
```

The result could be different? (not 100% sure)

Reply:
It shouldn’t (i changed it to avoid the force unwrap which triggers a SwiftLint violation)